libsupport/src/Strings.cpp

```diff
@@ -41,3 +41,41 @@ katana::TrimSuffix(const std::string& s, const std::string& suffix) {
   }
   return s;
 }
+
+std::string
+katana::FromBase64(std::string input) {
+  using namespace boost::archive::iterators;
+  typedef transform_width<
+      binary_from_base64<remove_whitespace<std::string::const_iterator>>, 8, 6>
+      ItBinaryT;
+
+  try {
+    // If the input isn't a multiple of 4, pad with =
+    size_t num_pad_chars((4 - input.size() % 4) % 4);
```

Comment:
I don't think the second `% 4` is necessary here.

Reply:
I think you are technically right but having second ```%4``` is useful for the case where the input size is a multiple of 4 and no padding is actually needed (the second ```%4``` pushes ```num_pad_chars``` back to zero).

Reply:
> how does this account for whitespace in `input`?

whitespace is not legal in Base64 encoding. If user passes in a base64 string with whitespace it will throw a boost data flow exception ( dataflow_exception(dataflow_exception::invalid_base64_character) ).